rs/nns/governance/src/governance.rs

```diff
@@ -2135,7 +2145,9 @@ impl Governance {
         self.remove_neuron(donor_neuron)?;
 
         self.with_neuron_mut(recipient_neuron_id, |recipient_neuron| {
-            recipient_neuron.cached_neuron_stake_e8s += transfer_amount_doms;
+            recipient_neuron.cached_neuron_stake_e8s = recipient_neuron
+                .cached_neuron_stake_e8s
+                .saturating_add(transfer_amount_doms);
```

Comment:
Ditto: why not checked_add?

In general, I think we want saturating_foo when either:

1. There is some nearby code that plainly ensures that there will be no overflow here, or
2. In the case of overflow, we actually want saturation for whatever reason.

It doesn't seem to me like either of these apply here. If a check is added before the transfer, then, we would be in case 1 (but we would have to lock the neurons to ensure that the check is still valid after the transfer).

I don't think we should use saturating_add as a less bad alternative to `+` (which wraps around in release builds).